# Practice notebook for confidence intervals using NHANES data

This notebook will give you the opportunity to practice working with confidence intervals using the NHANES data.

You can enter your code into the cells that say "enter your code here", and you can type responses to the questions into the cells that say "Type Markdown and Latex".

Note that most of the code that you will need to write below is very similar to code that appears in the case study notebook.  You will need to edit code from that notebook in small ways to adapt it to the prompts below.

To get started, we will use the same module imports and read the data in the same way as we did in the case study:

In [1]:
%matplotlib inline
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import seaborn as sns
import statsmodels.api as sm

da = pd.read_csv("nhanes_2015_2016.csv")

## Question 1

Restrict the sample to women between 35 and 50, then use the marital status variable [DMDMARTL](https://wwwn.cdc.gov/Nchs/Nhanes/2015-2016/DEMO_I.htm#DMDMARTL) to partition this sample into two groups - women who are currently married, and women who are not currently married.  Within each of these groups, calculate the proportion of women who have completed college.  Calculate 95% confidence intervals for each of these proportions.

In [91]:
# enter your code here
# females sample dropna?
da
gender = da[da['RIAGENDRx'] == 'Female']
females_sample = gender.sample(n=50)
females_sample['DMDMARTLx'] =females_sample.DMDMARTL.replace({1: "Married", 2: "Not_Married",2: "Not_Married",3: "Not_Married",4: "Not_Married", 5: "Not_Married", 6:"Not_Married", 77:"Not_Married", 99: "Not_Married"})

In [92]:
females_sample['DMDEDUC2x'] = females_sample.DMDEDUC2.replace({1:"Not Graduated", 2:"Not Graduated", 3:"Not Graduated", 4:"Not Graduated", 5:"Graduated", 7:"Not Graduated",9: "Not Graduated"}) 


In [93]:
fdf = females_sample[["DMDEDUC2x","DMDMARTLx"]].dropna()
fdf 
pd.crosstab(fdf.DMDEDUC2x,fdf.DMDMARTLx )

DMDMARTLx      Married  Not_Married
DMDEDUC2x                          
Graduated            9            5
Not Graduated       14           19

In [94]:
fdz = fdf.groupby(fdf.DMDMARTLx).agg({"DMDEDUC2x": [lambda x: np.mean(x=="Graduated"), np.size]})
fdz.columns = ["Proportion", "Total"] # The default column names are unclear, so we replace them here
fdz

Proportion  Total
DMDMARTLx                     
Married        0.391304     23
Not_Married    0.208333     24

In [95]:
p = fdz.Proportion.Married 
n = fdz.Total.Married 
se_Married = np.sqrt(p * (1 - p) / n)
print(se_Married)

p = fdz.Proportion.Not_Married 
n = fdz.Total.Not_Married 
se_Not_Married = np.sqrt(p * (1 - p) / n)
print(se_Not_Married)

0.10176384751376008
0.08289816934939807


In [132]:
p = fdz.Proportion.Married 
n = fdz.Total.Married 
lcbM= p - 1.96 * np.sqrt(p * (1 - p) / n)  
ucbM = p + 1.96 * np.sqrt(p * (1 - p) / n)  
print(lcbM, ucbM)
print(lcbM-ucbM)


0.19184720669911723 0.5907614889530567
-0.39891428225393943


In [133]:

p = fdz.Proportion.Not_Married 
n = fdz.Total.Not_Married 
lcbNM = p - 1.96 * np.sqrt(p * (1 - p) / n)  
ucbNM = p + 1.96 * np.sqrt(p * (1 - p) / n)  
print(lcbNM, ucbNM)
print(lcbNM-ucbNM)

0.04585292140851313 0.3708137452581536
-0.3249608238496404


__Q1a.__ Identify which of the two confidence intervals is wider, and explain why this is the case. 

Confidence interval for the Married females in wider as opposed to the not Married females. There is a larger proportion of the married females who have completed a college course relative to those who are not married. 

__Q1b.__ Write 1-2 sentences summarizing these findings for an audience that does not know what a confidence interval is (the goal here is to report the substance of what you learned about how marital status and educational attainment are related, not to teach a person what a confidence interval is).

As seen with the figures above there is a 95% likelihood that 39.1% of the females who are married have graduated with a college degree. That is to say if we ran this test mulitple times 95 times out of 100 we would expect to see such results. The same is for females who are not married and are college graduates. 

## Question 2

Construct 95% confidence intervals for the proportion of smokers who are female, and for the proportion of smokers who are male.  Then construct a 95% confidence interval for the difference between these proportions.

In [104]:
# enter your code here
da["SMQ020x"] = da.SMQ020.replace({1: "Yes", 2: "No", 7: np.nan, 9: np.nan})  # np.nan represents a missing value
dx = da[["SMQ020x", "RIAGENDRx"]].dropna()  # dropna drops cases where either variable is missing
pd.crosstab(dx.SMQ020x, dx.RIAGENDRx)


RIAGENDRx  Female  Male
SMQ020x                
No           2066  1340
Yes           906  1413

In [105]:
dz = dx.groupby(dx.RIAGENDRx).agg({"SMQ020x": [lambda x: np.mean(x=="Yes"), np.size]})
dz.columns = ["Proportion", "Total_n"] # The default column names are unclear, so we replace them here
dz

Proportion  Total_n
RIAGENDRx                     
Female       0.304845     2972
Male         0.513258     2753

In [106]:
p = dz.Proportion.Female # Female proportion
n = dz.Total_n.Female # Total number of females
se_female = np.sqrt(p * (1 - p) / n)
print(se_female)

p = dz.Proportion.Male # Male proportion
n = dz["Total_n"].Male # Total number of males
se_male = np.sqrt(p * (1 - p) / n)
print(se_male)

0.008444152146214435
0.009526078653689868


In [107]:
p = dz.Proportion.Female # Female proportion
n = dz.Total_n.Female # Total number of females
lcb = p - 1.96 * np.sqrt(p * (1 - p) / n)  
ucb = p + 1.96 * np.sqrt(p * (1 - p) / n)  
print(lcb, ucb)

0.288294683866098 0.32139576027925865


In [108]:
p = dz.Proportion.Male # Male proportion
n = dz.Total_n.Male # Total number of males
lcb = p - 1.96 * np.sqrt(p * (1 - p) / n)  
ucb = p + 1.96 * np.sqrt(p * (1 - p) / n)  
print(lcb, ucb)

0.49458714955108174 0.531929377873546


__Q2a.__ Discuss why it may be relevant to report the proportions of smokers who are female and male, and contrast this to reporting the proportions of males and females who smoke.

It is important to distinguish between the two genders as they will skew the results if combined. It will not be a true representation or near true representation of the real world.

__Q2b.__ How does the width of the confidence interval for the difference of the two proportions compare to the widths of the confidence intervals for each proportion separately?

## Question 3

Construct a 95% interval for height ([BMXHT](https://wwwn.cdc.gov/Nchs/Nhanes/2015-2016/BMX_I.htm#BMXHT)) in centimeters.  Then convert height from centimeters to inches by dividing by 2.54, and construct a 95% confidence interval for height in inches.  Finally, convert the endpoints (the lower and upper confidence limits) of the confidence interval from inches to back to centimeters   

In [112]:
# enter your code here
px = da[["BMXHT", "RIAGENDRx"]].dropna()  # dropna drops cases where either variable is missing
pd.crosstab(px.RIAGENDRx,px.BMXHT)


BMXHT      129.7  136.5  137.4  137.6  137.9  138.4  139.0  139.6  139.9  \
RIAGENDRx                                                                  
Female         1      0      1      1      1      1      2      1      1   
Male           0      1      0      0      0      0      0      0      0   

BMXHT      140.0  ...  193.7  193.8  194.3  194.6  195.1  195.4  195.6  198.4  \
RIAGENDRx         ...                                                           
Female         1  ...      0      0      0      0      0      0      0      0   
Male           0  ...      1      1      1      1      1      1      1      1   

BMXHT      201.0  202.7  
RIAGENDRx                
Female         0      0  
Male           1      1  

[2 rows x 509 columns]

In [116]:

pz = px.groupby(px.RIAGENDRx).agg({"BMXHT":np.mean})
pz


BMXHT
RIAGENDRx            
Female     159.673184
Male       173.132050

In [122]:
#centimeters to inches
da["BMXHTx"] = da.BMXHT/2.54
pix = da[["BMXHTx", "RIAGENDRx"]].dropna()  # dropna drops cases where either variable is missing
pd.crosstab(pix.RIAGENDRx,pix.BMXHTx)


BMXHTx     51.062992  53.740157  54.094488  54.173228  54.291339  54.488189  \
RIAGENDRx                                                                     
Female             1          0          1          1          1          1   
Male               0          1          0          0          0          0   

BMXHTx     54.724409  54.960630  55.078740  55.118110  ...  76.259843  \
RIAGENDRx                                              ...              
Female             2          1          1          1  ...          0   
Male               0          0          0          0  ...          1   

BMXHTx     76.299213  76.496063  76.614173  76.811024  76.929134  77.007874  \
RIAGENDRx                                                                     
Female             0          0          0          0          0          0   
Male               1          1          1          1          1          1   

BMXHTx     78.110236  79.133858  79.803150  
RIAGENDRx                                   
Female             0          0          0  
Male               1          1          1  

[2 rows x 509 columns]

In [123]:
piz = pix.groupby(pix.RIAGENDRx).agg({"BMXHTx":np.mean})
piz


BMXHTx
RIAGENDRx           
Female     62.863458
Male       68.162224

In [125]:
#CI for cm
female_height = da.loc[da.RIAGENDRx=="Female", "BMXHT"].dropna()
sm.stats.DescrStatsW(female_height).zconfint_mean()

(159.41341564918525, 159.932952307366)

In [126]:
male_height = da.loc[da.RIAGENDRx=="Male", "BMXHT"].dropna()
sm.stats.DescrStatsW(male_height).zconfint_mean()

(172.83801745601892, 173.42608228728875)

In [129]:
#CI for inches
female_height_in = da.loc[da.RIAGENDRx=="Female", "BMXHTx"].dropna()
sm.stats.DescrStatsW(female_height_in).zconfint_mean()

(62.761187263458766, 62.965729254868506)

In [131]:
male_height_in = da.loc[da.RIAGENDRx=="Male", "BMXHTx"].dropna()
sm.stats.DescrStatsW(male_height_in).zconfint_mean()

(68.04646356536176, 68.27798515247588)

__Q3a.__ Describe how the confidence interval constructed in centimeters relates to the confidence interval constructed in inches.

They should be similar as the one in inches is just a scalar division of 2.54 that of the on in cm

## Question 4

Partition the sample based on 10-year age bands, i.e. the resulting groups will consist of people with ages from 18-28, 29-38, etc. Construct 95% confidence intervals for the difference between the mean BMI for females and for males within each age band.

In [ ]:
# enter your code here

__Q4a.__ How do the widths of these confidence intervals differ?  Provide an explanation for any substantial diferences in the confidence interval widths that you see.

## Question 5

Construct a 95% confidence interval for the first and second systolic blood pressure measures, and for the difference between the first and second systolic blood pressure measurements within a subject.

In [ ]:
# enter code here

__Q5a.__ Based on these confidence intervals, would you say that a difference of zero between the population mean values of the first and second systolic blood pressure measures is consistent with the data?


__Q5b.__ Discuss how the width of the confidence interval for the within-subject difference compares to the widths of the confidence intervals for the first and second measures.

## Question 6

Construct a 95% confidence interval for the mean difference between the average age of a smoker, and the average age of a non-smoker.

In [ ]:
# insert your code here

__Q6a.__ Use graphical and numerical techniques to compare the variation in the ages of smokers to the variation in the ages of non-smokers.  

In [1]:
# insert your code here

__Q6b.__ Does it appear that uncertainty about the mean age of smokers, or uncertainty about the mean age of non-smokers contributed more to the uncertainty for the mean difference that we are focusing on here?